### DISCUSSION TITLE
[BP#4002] Questionnement sur le support de webp, ainsi que le format APNG

### DISCUSSION THREAD

**Discussion by Julien-Elie** [cnumr/best-practices#540] [Bonnes pratiques]
> Le webp animé est moindre dans son gain de poids et est actuellement peu supporté par les navigateurs.

J'ai l'impression que ce format est bien supporté dans les versions actuelles des principaux navigateurs en 2024. La formulation de la phrase pourrait être modifiée.

Aussi, ne faudrait-il pas parler du format APNG ? Il est plus ancien que webp, davantage supporté par les navigateurs depuis le temps, et offre même [un meilleur niveau de compression](http://littlesvr.ca/apng/gif_apng_webp.html)...

**Comment by youenchene** [comment]
Pour apporter au débat, c'est un sujet que j'ai retravaillé à plusieurs reprises cette année :

Semaine dernière (chiffre mai 2024) : https://www.lewebvert.fr/blog/2024-09-12-wcoi/

Début d'année (chiffre décembre 2023) : https://www.lewebvert.fr/blog/2024-02-28-webp-compatibilite/

Dans tous les cas il faut faire attention aux biais des petits pourcentages. 0.5% dans le monde Android c'est 30 millions de terminaux...

PS: sur can I use webp et apng sont très proche.

**Reply by Julien-Elie** [reply]
Merci pour l'analyse. Il faudrait voir quel pourcentage la formulation "peu supporté par les navigateurs" sous-entend. Ce n'est pas de l'ordre de 99,5 % de compatibilité, de ma compréhension de la formulation.
Autant donner factuellement ce pourcentage de 2024 alors 😄 

Par ailleurs, APNG semble plus efficient que WebP d'après ce que j'ai pu en lire.

**Reply by youenchene** [reply]
Le 0.5% c'est pour l'exemple, voici les chiffres à jour.


Format | Can I Use - Taux d’incompatibilité | iOS - Taux d’incompatibilité | Android - Taux d’incompatibilité
-- | -- | -- | --
WebP - 2020 (Safari). | 4,31% | 3,44% | 0,30%
AVIF - Janvier 2024 (MS Edge) | 7,36% | 11,64% | (non fiable) 51,40%
JPEG XL - Future Date - (safari uniquement compatible) | 86,77% | 24,99% | 100%

Le point est de contre balancer l'impact énergétique avec les impacts d'obsolescence matérielle provoqué par le logiciel.

AVIF et JXL sont aussi efficient que webp. Mais si on contre balance avec la partie compatibilité/obsolescence, dans un axe d'éco-conception c'est plus compliqué que de les conseiller (le APNG est mieux positionné que ces formats au passage).

**Comment by rlemaire** [comment]
Je suis d'avis d'utiliser des formulations « future proof » du genre « utiliser le format d'image le plus léger », et si on donne des exemples, de fournir une année et un contexte d'utilisation comme « site grand public en 2024 » ou « application web pour un parc d'appareils android vieilissant »

**Reply by youenchene** [reply]
Site grand public en 2024 n'est-il pas égale à "application web pour un parc d'appareils android vieilissant".

(le biais des petits %).

**Reply by rlemaire** [reply]
je n'ai pas compris Youen ? Ce sont deux exemples différents.

**Reply by youenchene** [reply]
Je sur-interprète sans trop.

Quand je lis "site grand public en 2024" j'ai l'impression que tu exclus qu'il y ait des androids vieillissant.

**Comment by raphthebaud** [comment]
Au delà de la question technique autour du support format, cette BP devrait peut-être traitée avec la 39 et la 4003 ?
Ca ne serait pas pertinent de fusionner plusieurs de ces règles ou de leur fournir un chapeau commun ?

Le premier objectif c'est quand même limiter le nombre d'animations, l'autoplay, et les animations non porteuses de sens quelle que soit la technique utilisée...

**Reply by youenchene** [reply]
Cela fait sens de séparer les 2 :

- pas les mêmes profils/personas en face
- pas les mêmes actions correctives.

Mon opinion est qu'il vaut mieux un message spécialisé par profil pour mieux propager les pratiques.
